ComposePrefs3/src/main/java/com/jamal/composeprefs3/ui/prefs/ListPref.kt

```diff
@@ -56,9 +56,10 @@ fun ListPref(
     buttonColor: Color = MaterialTheme.colorScheme.primary,
     enabled: Boolean = true,
     entries: Map<String, String> = mapOf(), //TODO: Change to List?
+    sortedBy: Comparator<Pair<String,String>>? = null
 ) {
 
-    val entryList = entries.toList()
+    val entryList = if(sortedBy==null)  entries.toList() else entries.toList().sortedWith(sortedBy)
```

Comment:
I'm not convinced this is working as you expect. It seems to me like this would sort the list on every recomposition. what if you have multiple ListPrefs with big lists. Seems like it wouldn't be very performant. Can you check how often the sort happens and if it is on every recomposition, then how to avoid that.

Reply:
You're right, the sort logic should be moved to the dataRepository or dataSource side to avoid multiple recomputation